packages/regenesis-surgery/test/predeploy.spec.ts

```diff
@@ -1,34 +1,178 @@
-/* eslint-disable @typescript-eslint/no-empty-function */
-describe.skip('predeploys', () => {
+import { expect, env } from './setup'
+import { AccountType } from '../scripts/types'
+import { GenesisJsonProvider } from './provider'
+
+describe('predeploys', () => {
+  const predeploys = {
+    eth: [],
+    newNotEth: [],
+    noWipe: [],
+    wipe: [],
+    weth: [],
+  }
+  // Base genesis file only
+  let genesisStateProvider: GenesisJsonProvider
+  // Old sequencer state
+  let oldStateProvider: GenesisJsonProvider
+
+  before(async () => {
+    await env.init()
+    predeploys.eth = env.getAccountsByType(AccountType.PREDEPLOY_ETH)
+    predeploys.newNotEth = env.getAccountsByType(
+      AccountType.PREDEPLOY_NEW_NOT_ETH
+    )
+    predeploys.noWipe = env.getAccountsByType(AccountType.PREDEPLOY_NO_WIPE)
+    predeploys.wipe = env.getAccountsByType(AccountType.PREDEPLOY_WIPE)
+    predeploys.weth = env.getAccountsByType(AccountType.PREDEPLOY_WETH)
+
+    genesisStateProvider = new GenesisJsonProvider(
+      env.surgeryDataSources.genesis
+    )
+    oldStateProvider = new GenesisJsonProvider(
+      env.surgeryDataSources.configs.stateDumpFilePath
+    )
+  })
+
   describe('new predeploys that are not ETH', () => {
-    it('should have the exact state specified in the base genesis file', async () => {})
+    for (const [i, account] of predeploys.newNotEth.entries()) {
+      describe(`account ${i}/${predeploys.newNotEth.length} (${account.address})`, () => {
+        it('should have the exact state specified in the base genesis file', async () => {
+          const preBytecode = await genesisStateProvider.getCode(
+            account.address
+          )
+          const postBytecode = await env.postL2Provider.getCode(account.address)
+          expect(preBytecode).to.eq(postBytecode)
+
+          // check storage
```

Comment:
@tynes storage is a lot easier to check if we just use the objects from the dumps instead of getting each slot by key from the provider, so should we just find the account in the post-surgery state and compare what's at its `.storage` for each of these?

Reply:
You should be able to use `provider.getStorageAt`